pkg/vgmanager/vgmanager_controller.go

```diff
@@ -0,0 +1,208 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vgmanager
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/google/go-cmp/cmp"
+	lvmv1alpha1 "github.com/red-hat-storage/lvm-operator/api/v1alpha1"
+	"github.com/red-hat-storage/lvm-operator/controllers"
+	"github.com/red-hat-storage/lvm-operator/pkg/internal"
+	"github.com/topolvm/topolvm"
+	"github.com/topolvm/topolvm/lvmd"
+	lvmdCMD "github.com/topolvm/topolvm/pkg/lvmd/cmd"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	corev1helper "k8s.io/component-helpers/scheduling/corev1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	ControllerName = "vg-manager"
+)
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *VGReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	r.deviceAgeMap = newAgeMap(&wallTime{})
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&lvmv1alpha1.LVMCluster{}).
+		Complete(r)
+}
+
+type VGReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+	Log    logr.Logger
+	// map from KNAME of device to time when the device was first observed since the process started
+	deviceAgeMap *ageMap
+}
+
+func (r *VGReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	r.Log = log.FromContext(ctx).WithName(ControllerName)
+	r.Log.Info("reconciling", "lvmcluster", req)
+	res, err := r.reconcile(ctx, req)
+	if err != nil {
+		r.Log.Error(err, "reconcile error")
+	}
+	r.Log.Info("reconcile complete", "result", res)
+	return res, err
+
+}
+func (r *VGReconciler) reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	lvmCluster := &lvmv1alpha1.LVMCluster{}
+	err := r.Client.Get(ctx, req.NamespacedName, lvmCluster)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	r.Log.Info("listing block devices")
+	//  list block devices
+	blockDevices, badRows, err := internal.ListBlockDevices()
+	if err != nil {
+		msg := fmt.Sprintf("failed to list block devices: %v", err)
+		r.Log.Error(err, msg, "lsblk.BadRows", badRows)
+		return ctrl.Result{}, err
+	} else if len(badRows) > 0 {
+		r.Log.Error(err, "could not parse all the lsblk rows", "lsblk.BadRows", badRows)
+	}
+
+	existingLvmdConfig := &lvmdCMD.Config{}
+
+	// load lvmd config
+	lvmdConfig := &lvmdCMD.Config{
+		SocketName: topolvm.DefaultLVMdSocket,
```

Comment:
- Is the value of `topolvm.DefaultLVMdSocket` and what's supplied at https://github.com/red-hat-storage/lvm-operator/blob/7b5eb349b5fd22374b24b6b571238999c6c68e7b/controllers/topolvm_node.go#L277 to nodeplugin same?

Reply:
yes. Same. https://github.com/topolvm/topolvm/blob/af1ff7bb2620d492e50047a041997fd86d55ac48/constants.go#L36

Reply:
not same.  https://github.com/topolvm/topolvm/blob/af1ff7bb2620d492e50047a041997fd86d55ac48/constants.go#L36